swap/src/monero/wallet.rs

```diff
@@ -87,17 +87,23 @@ impl Wallet {
 
         // Properly close the wallet before generating the other wallet to ensure that
         // it saves its state correctly
-        let _ = wallet.close_wallet().await?;
+        let _ = wallet
+            .close_wallet()
+            .await
+            .context("Failed to close wallet")?;
 
         let _ = wallet
             .generate_from_keys(
                 file_name,
-                &address.to_string(),
-                &private_spend_key.to_string(),
-                &PrivateKey::from(private_view_key).to_string(),
+                address.to_string(),
+                private_spend_key.to_string(),
+                PrivateKey::from(private_view_key).to_string(),
                 restore_height.height,
+                String::from(""),
+                true,
```

Comment:
🙃 With this parameter `autosave_current` being set to `true` can remove `close_wallet`. Because we only close the wallet in cases where we generate a new wallet. 
( I had overlooked this parameter before :)

Reply:
I think you might be right. I want to keep this a pure refactor though if possible but happy to merge a follow-up PR :)